internal/gh/client.go

```diff
@@ -91,7 +101,17 @@ func (c *Client) restPost(
 	}
 
 	startTime := time.Now()
-	url := githubApiBaseUrl + endpoint
+
+	// GitHub cloud and GHES have different API URLs.
+	// For cloud, it's `https://api.github.com/repos/...`.
+	// For GHES, it's `https://github.mycompany.com/api/v3/repos/...`.
+	var url string
+	if config.Av.GitHub.BaseURL == "" {
+		url = githubCloudApiBaseUrl + endpoint
+	} else {
+		url = config.Av.GitHub.BaseURL + "/v3" + endpoint
```

Comment:
We do always need the `v3`: https://docs.github.com/en/enterprise-server@3.11/rest/enterprise-admin?apiVersion=2022-11-28#endpoint-urls.

The reason I didn't make the user add the `/v3` is because we use the same config variable for GraphQL, which should **not** have the `/v3`. i.e., rest endpoints are `github.mycompany.com/api/v3/repo/...`, GraphQL is `github.mycompany.com/api/graphql`.

Though I think now that you mention it, we should probably be adding the `/api/` bit to both places instead of making the user add it.

Reply:
For GitHub cloud (including enterprise cloud) users should not set this value because it should use `api.github.com`. It's only relevant for GHES, at which point, the API url will always be `github.mycorp.com/api/v3/repo/...` or `github.mycorp.com/api/graphql`.

I don't see a reason to add two config values for that since a user would have to set either both or none.